File: dpi-backend/src/main/java/com/sysinfo/dpi/patient/dto/PatientSearchCriteria.java
package com.sysinfo.dpi.patient.dto;

import com.sysinfo.dpi.patient.models.Sexe;
import java.time.LocalDate;
import lombok.Data;

/**
 * @author Ali Bouali
 * @since 17.04.22
 */

@Data
public class PatientSearchCriteria {

  private String nationalId;

  private String firstName;

  private String middleName;

  private String lastname;

  private String phoneNumber;

  private LocalDate birthdate;

  private String email;

  private Sexe sexe;

}

File: dpi-frontend/src/app/services/models/patient-search-criteria.ts
/* tslint:disable */
export interface PatientSearchCriteria {
  nationalId?: string;
  firstName?: string;
  middleName?: string;
  lastname?: string;
  phoneNumber?: string;
  birthdate?: string;
  email?: string;
  sexe?: 'MALE' | 'FEMALE';
}
